src/node_buffer.cc

```diff
@@ -1205,6 +1205,60 @@ void DetachArrayBuffer(const FunctionCallbackInfo<Value>& args) {
   }
 }
 
+// on success: returns a non-negative integer indicating the size of the
+// binary produced, it most be no larger than 2147483647 bytes.
+// In case of error, a negative value is returned:
+// * -1 indicates a single character remained,
+// * -2 indicates an invalid character,
+// * -3 indicates a possible overflow (i.e., more than 2 GB output).
+static void Atob(const FunctionCallbackInfo<Value>& args) {
+  CHECK_EQ(args.Length(), 1);
+  Environment* env = Environment::GetCurrent(args);
+  THROW_AND_RETURN_IF_NOT_STRING(env, args[0], "argument");
+
+  Local<String> input = args[0].As<String>();
+  MaybeStackBuffer<char> buffer;
+  simdutf::result result;
+
+  if (input->IsExternalOneByte()) {  // 8-bit case
+    auto ext = input->GetExternalOneByteStringResource();
+    size_t expected_length =
+        simdutf::maximal_binary_length_from_base64(ext->data(), ext->length());
+    buffer.AllocateSufficientStorage(expected_length + 1);
+    buffer.SetLengthAndZeroTerminate(expected_length);
+    result = simdutf::base64_to_binary(
+        ext->data(), ext->length(), buffer.out(), simdutf::base64_default);
+  } else {  // 16-bit case
+    String::Value value(env->isolate(), input);
+    auto data = reinterpret_cast<const char16_t*>(*value);
+    size_t expected_length =
+        simdutf::maximal_binary_length_from_base64(data, value.length());
+    buffer.AllocateSufficientStorage(expected_length + 1);
+    buffer.SetLengthAndZeroTerminate(expected_length);
+    result = simdutf::base64_to_binary(
+        data, value.length(), buffer.out(), simdutf::base64_default);
+  }
+
+  if (result.error == simdutf::error_code::SUCCESS) {
+    auto value =
+        String::NewFromOneByte(env->isolate(),
```

Comment:
There might be a problem if buffer.out() contains a null character... ? To be verified.

Reply:
@anonrig We have a failed test...
```
19:13:27     [UNEXPECTED_FAILURE][FAIL] btoa("\0a") == "AGE="
19:13:27     assert_equals: atob(btoa(input)) must be the same as String(input) expected "\0a" but got ""
```

I think that's because 'AGE=' goes to "\0a" which, as a null terminated string, is the empty string.

Reply:
`result.count` contains the length of the byte sequence, and that's what we need to use, I think.

Reply:
The complete function signature is...

```C++
NewFromOneByte(Isolate* isolate, const uint8_t* data,NewStringType type = NewStringType::kNormal, int length);
```

I think we should pass the length parameter (defaults on -1).